url3ed4973106_kosmasK_RecordPlayer_tgz-pJ8-es_uc3m_recordplayer_test_TestJ8/src/es/uc3m/eda/list/singlelink/SQueue.java

```diff
@@ -0,0 +1,66 @@
+package es.uc3m.eda.list.singlelink;
+
+import es.uc3m.eda.list.IQueue;
+
+public class SQueue<E> implements IQueue<E> {
+	private SNode<E> frontNode;
+	private SNode<E> tailNode;
+
+	@Override
+	public boolean isEmpty() {
+		return this.frontNode == null;
+	}
+
+	@Override
+	public void enqueue(E elem) {
+		SNode<E> node = new SNode<E>(elem);
+		if (this.tailNode == null) {
+			this.frontNode = node;
+		} else {
+			this.tailNode.nextNode = node;
+		}
+		this.tailNode = node;
+	}
+
+	@Override
+	public E dequeue() {
+		if (this.frontNode == null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 27 to != check before setting the field value on line 31 or verify if the check is needed at all.